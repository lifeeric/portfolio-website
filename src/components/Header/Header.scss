@import "../../components/Theme/variables";

.header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 100;
  background: $whiteColor;
  transition: 0.5s ease;
  box-shadow: 0px 2px 4px rgba(34, 36, 43, 0.16);

  &__center {
    display: flex;
    justify-content: center;
  }

  &__innerWidth {
    max-width: 1100px;
    width: 100%;
    margin: 0 25px;
  }

  &__wrapper {
    display: flex;
    justify-content: space-between;
    align-items: center;
    z-index: 100;
    padding: 15px 0;
  }

  &__logo {
    font-family: nunito, sans-serif;
    font-size: 30px;
    font-weight: 800;
    width: 70px;

    & > a {
      text-decoration: none;
      color: $primary;
    }
  }
}

@media (max-width: 600px) {
  .header {
    padding: 20px 0px;

    // wrapper
    &__wrapper {
      padding: 0;
    }
  }
}
