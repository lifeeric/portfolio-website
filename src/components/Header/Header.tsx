import * as React from "react"
import { Link, graphql, useStaticQuery } from "gatsby"
import Img from "gatsby-image"

import { Navbar } from "../Navbar/Navbar"

import "./Header.scss"

export const Header: React.FC = () => {
  const data = useStaticQuery(graphql`
    query {
      file(relativePath: { eq: "gatsby-icon.png" }) {
        childImageSharp {
          fluid {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `)

  console.log("IMAGE ", data)

  return (
    <header className="header">
      <div className="header__center">
        <div className="header__innerWidth">
          <div className="header__wrapper">
            <div className="header__logo">
              <Link to="/">
                <Img
                  fluid={data.file.childImageSharp.fluid}
                  alt="Eric Software Engineer || Hire expert Software engineer"
                />
              </Link>
            </div>
            <Navbar />
          </div>
        </div>
      </div>
    </header>
  )
}
