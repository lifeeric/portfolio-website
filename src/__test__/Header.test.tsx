import * as React from "react"
import { shallow, mount } from "enzyme"

import { Header } from "../components/Header/Header"

describe("<Header /> testing", () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = mount(<Header />)
  })
  it("should render well", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should render again", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
