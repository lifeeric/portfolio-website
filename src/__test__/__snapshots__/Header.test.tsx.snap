// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> testing should render again 1`] = `
<Component>
  <header
    className="header"
  >
    <div
      className="header__center"
    >
      <div
        className="header__innerWidth"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__logo"
          >
            el
          </div>
          <Component>
            <div
              className="toggle"
            >
              <div
                className="toggle_right"
                onClick={[Function]}
              >
                <div
                  className="toggle__dots"
                >
                  <div
                    className="toggle__dot--1"
                  />
                  <div
                    className="toggle__dot--2"
                  />
                  <div
                    className="toggle__dot--3"
                  />
                </div>
              </div>
            </div>
            <nav
              className="navbar false"
            >
              <span
                className="navbar__close"
                onClick={[Function]}
              />
              <ul
                className="navbar__list"
              >
                <Component
                  key="home"
                  name="Me"
                  offset={-120}
                  onClick={[Function]}
                  to="home"
                >
                  <li
                    className="navbar__item"
                  >
                    <Link
                      activeClass="active"
                      duration={500}
                      offset={-120}
                      onClick={[Function]}
                      smooth={true}
                      spy={true}
                      to="home"
                    >
                      <LinkElement
                        onClick={[Function]}
                      >
                        <a
                          onClick={[Function]}
                        >
                          Me
                        </a>
                      </LinkElement>
                    </Link>
                  </li>
                </Component>
                <Component
                  key="project"
                  name="Projects"
                  offset={-70}
                  onClick={[Function]}
                  to="project"
                >
                  <li
                    className="navbar__item"
                  >
                    <Link
                      activeClass="active"
                      duration={500}
                      offset={-70}
                      onClick={[Function]}
                      smooth={true}
                      spy={true}
                      to="project"
                    >
                      <LinkElement
                        onClick={[Function]}
                      >
                        <a
                          onClick={[Function]}
                        >
                          Projects
                        </a>
                      </LinkElement>
                    </Link>
                  </li>
                </Component>
                <Component
                  key="skill"
                  name="Skills"
                  offset={-70}
                  onClick={[Function]}
                  to="skill"
                >
                  <li
                    className="navbar__item"
                  >
                    <Link
                      activeClass="active"
                      duration={500}
                      offset={-70}
                      onClick={[Function]}
                      smooth={true}
                      spy={true}
                      to="skill"
                    >
                      <LinkElement
                        onClick={[Function]}
                      >
                        <a
                          onClick={[Function]}
                        >
                          Skills
                        </a>
                      </LinkElement>
                    </Link>
                  </li>
                </Component>
                <Component
                  key="contact"
                  name="Contact"
                  offset={-70}
                  onClick={[Function]}
                  to="contact"
                >
                  <li
                    className="navbar__item"
                  >
                    <Link
                      activeClass="active"
                      duration={500}
                      offset={-70}
                      onClick={[Function]}
                      smooth={true}
                      spy={true}
                      to="contact"
                    >
                      <LinkElement
                        onClick={[Function]}
                      >
                        <a
                          onClick={[Function]}
                        >
                          Contact
                        </a>
                      </LinkElement>
                    </Link>
                  </li>
                </Component>
              </ul>
            </nav>
          </Component>
        </div>
      </div>
    </div>
  </header>
</Component>
`;

exports[`<Header /> testing should render well 1`] = `
<Component>
  <header
    className="header"
  >
    <div
      className="header__center"
    >
      <div
        className="header__innerWidth"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__logo"
          >
            el
          </div>
          <Component>
            <div
              className="toggle"
            >
              <div
                className="toggle_right"
                onClick={[Function]}
              >
                <div
                  className="toggle__dots"
                >
                  <div
                    className="toggle__dot--1"
                  />
                  <div
                    className="toggle__dot--2"
                  />
                  <div
                    className="toggle__dot--3"
                  />
                </div>
              </div>
            </div>
            <nav
              className="navbar false"
            >
              <span
                className="navbar__close"
                onClick={[Function]}
              />
              <ul
                className="navbar__list"
              >
                <Component
                  key="home"
                  name="Me"
                  offset={-120}
                  onClick={[Function]}
                  to="home"
                >
                  <li
                    className="navbar__item"
                  >
                    <Link
                      activeClass="active"
                      duration={500}
                      offset={-120}
                      onClick={[Function]}
                      smooth={true}
                      spy={true}
                      to="home"
                    >
                      <LinkElement
                        onClick={[Function]}
                      >
                        <a
                          onClick={[Function]}
                        >
                          Me
                        </a>
                      </LinkElement>
                    </Link>
                  </li>
                </Component>
                <Component
                  key="project"
                  name="Projects"
                  offset={-70}
                  onClick={[Function]}
                  to="project"
                >
                  <li
                    className="navbar__item"
                  >
                    <Link
                      activeClass="active"
                      duration={500}
                      offset={-70}
                      onClick={[Function]}
                      smooth={true}
                      spy={true}
                      to="project"
                    >
                      <LinkElement
                        onClick={[Function]}
                      >
                        <a
                          onClick={[Function]}
                        >
                          Projects
                        </a>
                      </LinkElement>
                    </Link>
                  </li>
                </Component>
                <Component
                  key="skill"
                  name="Skills"
                  offset={-70}
                  onClick={[Function]}
                  to="skill"
                >
                  <li
                    className="navbar__item"
                  >
                    <Link
                      activeClass="active"
                      duration={500}
                      offset={-70}
                      onClick={[Function]}
                      smooth={true}
                      spy={true}
                      to="skill"
                    >
                      <LinkElement
                        onClick={[Function]}
                      >
                        <a
                          onClick={[Function]}
                        >
                          Skills
                        </a>
                      </LinkElement>
                    </Link>
                  </li>
                </Component>
                <Component
                  key="contact"
                  name="Contact"
                  offset={-70}
                  onClick={[Function]}
                  to="contact"
                >
                  <li
                    className="navbar__item"
                  >
                    <Link
                      activeClass="active"
                      duration={500}
                      offset={-70}
                      onClick={[Function]}
                      smooth={true}
                      spy={true}
                      to="contact"
                    >
                      <LinkElement
                        onClick={[Function]}
                      >
                        <a
                          onClick={[Function]}
                        >
                          Contact
                        </a>
                      </LinkElement>
                    </Link>
                  </li>
                </Component>
              </ul>
            </nav>
          </Component>
        </div>
      </div>
    </div>
  </header>
</Component>
`;
